tests/unit/src/main/scala/tests/CustomInputExpectSuite.scala

```diff
@@ -0,0 +1,71 @@
+package tests
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import tests.BuildInfo.testResourceDirectory
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.metals.RecursivelyDelete
+import scala.meta.io.AbsolutePath
+
+/**
+ * Suite for expect-tests with custom input files. Expects the following structure:
+ * <p>- resources/[[directoryName]]/input</p>
+ * <p>- resources/[[directoryName]]/expect</p>
+ */
+abstract class CustomInputExpectSuite(directoryName: String)
```

Comment:
Can we reuse `DirectoryExpectSuite` instead? It's not necessary to use the same input directory as the other expect suites, we can create a custom `InputProperties`

```scala
override def testCases(): List[ExpectTestCase] =  {
  val dir = AbsolutePath(testResourceDirectory).resolve(directoryName) // ???
  val customInput = InputProperties(dir, List(dir), Classpath(Nil), Classpath(Nil))
  customInput.allFiles.map { file =>
      ExpectTestCase(file, { () => obtainFrom(...) })
  }
```

That would avoid the need for `CustomInputExpectSuite`.

Reply:
It fails due to the directory structure.
```
testSuite/inputs/foo.scala
testSuite/expect/foo.scala
```

Directory structure requires outputs to mirror the structure from inputs:
```
testSuite/inputs/foo.scala
testSuite/foo.scala  // there is no `expect` directory
```

Reply:
Can you elaborate? The output directory needs to mirror the input directory but you can provide a custom input directory, you don't have to use `tests/input/src/main/` as input.

Reply:
OK, nevermind - I was wrong

Got it working using:
`extends DirectoryExpectSuite("foldingRange/expect")`
and 
`val inputDirectory = AbsolutePath(testResourceDirectory).resolve("foldingRange").resolve("input")`